ir/shared/src/main/scala/org/scalajs/ir/Trees.scala

```diff
@@ -1088,7 +1088,7 @@ object Trees {
 
   // Atomic expressions
 
-  sealed case class VarRef(ident: LocalIdent)(val tpe: Type)(
+  sealed case class VarRef(name: LocalName)(val tpe: Type)(
```

Comment:
Note: I have briefly considered whether `VarRef` should extend `LocalIdent` (or even whether we should remove `LocalIdent` altogether and use `VarRef` instead.

I do not think we should do this:
- VarDef / ParamDef could lose their `vtpe` / `ptpe` field, but it feels unnatural to have the types in the name.
- Other trees `ForIn` currently do not require explicit types, so we'd force an unnecessary `AnyType` there (which we'd have to IR Check).